src/Microsoft.AspNet.Mvc.Core/Filters/TypeFilterAttribute.cs

```diff
@@ -25,18 +24,8 @@ public TypeFilterAttribute([NotNull] Type type)
 
         public IFilter CreateInstance([NotNull] IServiceProvider serviceProvider)
         {
-            var activator = serviceProvider.GetRequiredService<ITypeActivator>();
-            var obj = activator.CreateInstance(serviceProvider, ImplementationType, Arguments ?? new object[0]);
-
-            var filter = obj as IFilter;
-            if (filter == null)
-            {
-                throw new InvalidOperationException(Resources.FormatFilterFactoryAttribute_TypeMustImplementIFilter(
-                    typeof(TypeFilterAttribute).Name,
-                    typeof(IFilter).Name));
-            }
-
-            return filter;
+            var filterActivator = serviceProvider.GetRequiredService<IFilterActivator>();
+            return filterActivator.CreateInstance(ImplementationType);
```

Comment:
Generally a `TypeFilterAttribute` is going to have application-scope. There's no need for a separate cache